ledger/sandbox/src/main/resources/db/migration/postgres/V34__sequential_row_id.sql

```diff
@@ -0,0 +1,26 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+---------------------------------------------------------------------------------------------------
+-- V34: Add a sequential row_id to participant_events
+--
+---------------------------------------------------------------------------------------------------
+
+-- 1. add the column
+alter table participant_events add column row_id serial;
+
+-- 2. fix the row_id to be sequential according to the order of (event_offset, transaction_id, node_index)
+update participant_events
+set row_id = t.rownum
+from (select event_offset, node_index, row_number() over (order by event_offset, transaction_id, node_index) as rownum
+      from participant_events) t
+where participant_events.event_offset = t.event_offset and participant_events.node_index = t.node_index;
+
+-- 3. drop the now unused index
+drop index participant_events_event_offset_transaction_id_node_index_idx;
+
+-- 4. create a new index involving row_id
+create index on particpant_events (row_id, event_offset, node_index);
```

Comment:
```suggestion
create index on participant_events (row_id);
```

Reply:
this would also speed up the inserts and should works fine with the query:
```
select * from participant_events where row_id >= 903006 order by row_id LIMIT 1000;
```